Window Swapping Refactor and CSV File Export Works

rhorton5 opened the pull request:
# WindowChanger.cs is removed
- It is replaced with WindowSwapper.cs.  This was to keep the naming scheme consistent with what we have been calling the method to swapping windows.

# WindowSwapper.cs is added 

- This is intended to manage Swapping Windows and determining whether or not the program is going to close the program entirely or just swap to a new window.  This minimizes access to `Program.cs` by having another class be responsible for swapping classes.

- *Note*: All new classes would require to have a `FormClosing` event to allow `WindowSwapper.ValidateWindow(this.Name)` to be added in the code.  The rest is automated by the class itself.
Changing Window just requires a class based on this `WindowSwapper.GoToYourFormClassNameHere(this)` (i.e. `WindowSwapper.GoToSpecimensForm(this)`).

# CSVExporter.cs is added

- This requires a single class that takes the DataGridView of the program and generates the csv format.  Should this be used in another program all you just need to do is `CSVExporter.Export(yourDataGridViewGoesHere)`  (i.e. `CSVExporter.Export(specimensDataGridView)`).

- `ExportToCSV(String data)` moved from `SpecimenForms.cs` to this class.  It is kept private as this will always occur after `Export(DataGridView dataGridView)`.

# Program.cs

- Removed arbitrary global variables.

# New Arrow Added For All NavigatorBinding / ToolStrip
- All NavigatorBinding and ToolStrips have a new Arrow that is more visible for the button that allows users to change to a specific window.

- *Note*: NavigatorBinding appears to be discontinued from .NET 6, which is why `SpecimensSummaryForm.cs` has a ToolStrip instead.  ToolStrip does not include the same functionalities as NavigatorBinding did.  This might be something that needs to be looked into in the future.

Commit:
Window Swapping Refactor and CSV File Export Works

NicholasJohnsonDev commented:
Starting my review

NicholasJohnsonDev commented on Team 6 Senior Project/CSVExporter.cs:
```diff
@@ -0,0 +1,51 @@
+                    Console.WriteLine(ex.Message);
+                }
+            }
+        }
```
close the SaveFileDialog with a dispose after your if. Between lines 32 and 33:
save.Dispose();

rhorton5 replied:
Went ahead and added save.Dispose() after the if statement.  I changed up the conditions for the if statement so it wouldn't display an error whenever you clicked cancel or exited out without saving on the save window.

NicholasJohnsonDev replied:
I was just staring at that if trying to figure it out.

NicholasJohnsonDev commented:
Approved. My comment on CSVExporter is optional as the garbage collector should take care of that anyway. I'll let you do the merge and judge if that change is necessary. 
The swapper and the exporter look so much cleaner in all the classes that use them!

Commit:
SaveDialog.Dispose() added and minor bug fix

SaveDialog now disposes after being used and fixed a small bug where an error message would display when you pressed "Cancel" on the save window.

rhorton5 commented:
I will merge this!  Just wanted to do some minor fixes on the CSVExporter before merging.

NicholasJohnsonDev commented:
Thank you, Ryley!
